infrastructure/templates/public-api/main.bicep

```diff
@@ -107,14 +127,50 @@ module applicationInsightsModule 'components/appInsights.bicep' = {
   }
 }
 
+// Create a generic, shared Log Analytics Workspace for any relevant resources to use.
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {
+  name: '${subscription}-ees-log'
+  location: location
+  properties: {
+    sku: {
+      name: 'PerGB2018'
+    }
+  }
+  tags: tagValues
+}
+
+// Create a generic Container App Environment for any Container Apps to use.
+resource containerAppEnvironment 'Microsoft.App/managedEnvironments@2023-05-01' = {
+  name: '${subscription}-ees-cae'
+  location: location
+  properties: {
+    vnetConfiguration: {
+      infrastructureSubnetId: vNetModule.outputs.containerAppEnvironmentSubnetRef
+    }
+    daprAIInstrumentationKey: applicationInsightsModule.outputs.applicationInsightsKey
+    appLogsConfiguration: {
+      destination: 'log-analytics'
+      logAnalyticsConfiguration: {
+        customerId: logAnalyticsWorkspace.properties.customerId
+        sharedKey: logAnalyticsWorkspace.listKeys().primarySharedKey
+      }
+    }
+    workloadProfiles: [{
+      name: 'Consumption'
+      workloadProfileType: 'Consumption'
+    }]
+  }
+  tags: tagValues
+}
```

Comment:
You could move these to the `components` folder as new files like `containerAppEnvironment.bicep` and `monitoring.bicep`, parameterising them and including them back here as modules. It wouldn't save that many lines in `main.bicep` right now but I don't think we'd want a trend of defining all new resources inline in `main.bicep` which could eventually lead to this file becoming much bigger than it needs to be.

Reply:
I have broken these down into modules now with various default values, and it has actually saved a good chunk of space so far (so long as we're using the defaults!)